clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaBasedCommandSender.java

```diff
@@ -157,4 +290,42 @@ public KafkaBasedCommandSender(
 
         return props;
     }
+
+    private Future<Void> subscribeForCommandResponse(final String tenantId,
+            final Handler<DownstreamMessage<KafkaMessageContext>> messageHandler, final Span span) {
+        if (commandResponseSubscriptions.get(tenantId) != null) {
+            span.log("command response consumer already exists");
+            LOGGER.debug("command response consumer already exists for tenant[{}]", tenantId);
+            return Future.succeededFuture();
+        } else {
+            final Map<String, String> consumerConfig = this.consumerConfig
```

Comment:
Does `KafkaBasedCommandSender` make any assumptions about the value of [https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#consumerconfigs_auto.offset.reset](auto.offset.reset)?

In the unit test I saw this is being set to `earliest`. The default value however is `latest`. Independent from that it can be set by configuration of the Kafka consumer, right?

Hence, if this class makes an assumption on the value of `auto.offset.reset` I guess it makes sense to make sure it can at least not be misconfigured.

Reply:
The command responses are correlated using the correlation id. Once a response has been correlated then corresponding entry has been removed from the list. If the same response is received again then it is simply ignored. Setting `auto.offset.reset` to `earliest` or `latest` doesn't change the behavior of the implementation. 

@fkaltner As we discussed, setting to `earliest` could possibly lead to some overhead when the north bound application is restarted. Then the consumer receives responses that have been correlated already, tries to correlate them again and at the end ignore them. In order to avoid this, we can override in the code that the value of the `auto.offset.reset`is always `latest`.@calohmn WDYT?

Reply:
I agree, better to explicitly set `auto.offset.reset` to `latest` here.